lib/oauth2/strategy/base.rb

```diff
@@ -11,6 +11,14 @@ def initialize(client)
       def client_params
         {'client_id' => @client.id, 'client_secret' => @client.secret}
       end
+
+      # Returns the Authorization header value for Basic Authentication
+      #
+      # @param [String] The client ID
+      # @param [String] the client secret
+      def authorization(client_id, client_secret)
+        'Basic ' + Base64.encode64(client_id + ':' + client_secret).gsub("\n", '')
```

Comment:
In my experiments with authenticating with a few OAuth providers, it seems that at least some reject form-encoded credentials. I have one with a `!` in the password, which would get encoded as `%21`, but this is rejected.